### PR TITLE
Add tags to indices for resources, files, schemas

### LINKED ISSUE #943: Add tag metadata to the indices
The tag metadata is not added to `elasticsearch` and `sparql` indices.

We should only introduce it for resources/schemas and files

### PR DESCRIPTION
**Pull request by dantb**
Fixes #943 

I checked that the mapping is as expected my local ES.

Unsure if / where I need to add tests? `ListingsSpec` hits the metadata field (hence the change)

### CODE REVIEW

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala [hunk 1]
```diff
@@ -37,32 +41,44 @@ final case class File(
     storageType: StorageType,
     attributes: FileAttributes,
     tags: Tags
-)
+) {
+  def metadata(rev: Int): Metadata = Metadata(tags.byRevision(rev))
```
Comment (imsdu) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala:45]: We want all tags

**Review by imsdu — CHANGES_REQUESTED** [review]

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala [hunk 2]
```diff
@@ -37,32 +41,42 @@ final case class File(
...
-      val storageJson                       = Json.obj(
-        keywords.id  -> file.storage.iri.asJson,
-        keywords.tpe -> storageType.iri.asJson,
-        "_rev"       -> file.storage.rev.asJson
-      )
-      file.attributes.asJsonObject.add("_storage", storageJson)
-    }
+  final case class Metadata(tags: List[UserTag])
+
+  implicit def fileEncoder(implicit config: StorageTypeConfig): Encoder[File] = { file =>
```
Comment (dantb) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala:52]: I noticed we're doing `Encoder.AsObject` all over the place - I'm not sure this is necessary? The `Json` ADT provides functions to work with objects (`mapObject` etc). It's designed so we operate on the abstraction - I think this adds unnecessary boilerplate?

Comment (imsdu) [reply delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/File.scala:52]: It does look nicer this way

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/schemas/model/Schema.scala [hunk 3]
```diff
@@ -88,14 +94,21 @@ object Schema {
         value.source.topContextValueOrEmpty.merge(ContextValue(contexts.shacl))
     }
 
-  type Shift = ResourceShift[SchemaState, Schema, Nothing]
+  implicit private val fileMetadataEncoder: Encoder[Metadata] = { m =>
+    Json.obj("_tags" -> m.tags.asJson)
+  }
+
+  implicit val fileMetadataJsonLdEncoder: JsonLdEncoder[Metadata] =
+    JsonLdEncoder.computeFromCirce(ContextValue(contexts.schemasMetadata))
```
Comment (imsdu) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/schemas/model/Schema.scala:102]: `contexts.metadata`

**Review by imsdu — APPROVED** [review]

**Pull request merged**